### PR TITLE
csi: SSH only to first reachable host in volume['g_host'] entry

### LINKED ISSUE #621: Unable to set quota-deem-statfs via ssh
Unable to set quota-deem-statfs via ssh when have more 1 gluster_hosts:

Storage with 3 hosts
```
apiVersion: kadalu-operator.storage/v1alpha1
kind: KadaluStorage
metadata:
  name: ext-gfs3
  namespace: default
spec:
  details:
    gluster_hosts:
      - 10.97.145.111
      - 10.97.145.112
      - 10.97.145.113
    gluster_options: log-level=DEBUG
    gluster_volname: kadaul
  kadalu_format: native
  pvReclaimPolicy: delete
  type: External
```
kadalu-provision:
```
[2021-08-17 11:47:21,746] INFO [kadalulib - 339:start_all] - Started Process	 name=csi
[2021-08-17 11:47:21,749] INFO [kadalulib - 339:start_all] - Started Process	 name=metrics
[2021-08-17 11:47:21,752] INFO [kadalulib - 339:start_all] - Started Process	 name=volumewatch
[2021-08-17 11:47:21,755] INFO [kadalulib - 339:start_all] - Started Process	 name=quota
Starting the quota crawler script
No PVC yet, continuing to watch...
Starting watch on configmap
Setting up watches.
Watches established.
[2021-08-17 11:47:21,873] INFO [exporter - 85:<module>] - Started Kadalu Storage CSI Metrics exporter.	 port=8000
[2021-08-17 11:47:21,981] ERROR [volumeutils - 1019:mount_glusterfs] - Unable to set quota-deem-statfs via ssh	 error=[255]  ssh: Could not resolve hostname 10.97.145.111,10.97.145.112,10.97.145.113: Name or service not known
[2021-08-17 11:47:21,982] ERROR [main - 42:mount_storage] - Unable to mount volume	 hvol=ext-gfs3
[2021-08-17 11:47:21,982] INFO [main - 43:mount_storage] - Volume is mounted successfully	 hvol=ext-gfs3
```

Need add connect to any host in server list.

### PR DESCRIPTION
**Pull request by vatsa287**
Fixes: #621 
Signed-off-by: Shree Vatsa N <vatsa@kadalu.io>

### CODE REVIEW

File: csi/volumeutils.py [hunk 1]
```diff
@@ -1019,7 +1023,7 @@ def mount_glusterfs(volume, mountpoint, is_client=False):
                 "-oStrictHostKeyChecking=no",
                 "-i",
                 "%s" % secret_private_key,
-                "%s@%s" % (secret_username, volume['g_host']),
+                "%s@%s" % (secret_username, g_host),
```
Comment (amarts) [inline csi/volumeutils.py:1030]: it is used in another place also? during volume expansion. Can you check if thats handled too?

Also mention Issue Reference in commit message.

File: csi/volumeutils.py [hunk 2]
```diff
@@ -1010,6 +1010,10 @@ def mount_glusterfs(volume, mountpoint, is_client=False):
             use_gluster_quota = True
         secret_private_key = "/etc/secret-volume/ssh-privatekey"
         secret_username = os.environ.get('SECRET_GLUSTERQUOTA_SSH_USERNAME', None)
+        
+        # SSH into only first host in volume['g_host'] entry
+        g_host = volume['g_host'].split(',')[0]
```
Comment (aravindavk) [inline csi/volumeutils.py]: Select a host that is reachable when multiple hosts provided.

For example,

```
def reachable_host(hosts):
    for host in hosts:
        host = host.strip()
        if is_host_reachable(host, 22):
            return host

    return None

g_host = reachable_host(volume['g_host'].strip().split(','))
if g_host is None:
    # Error, All hosts are not reachable
```

Comment (vatsa287) [reply csi/volumeutils.py]: is_host_reachable() is defined in operartor/main.py should I import it or re-define.

Comment (aravindavk) [reply csi/volumeutils.py]: You can move that function to kadalulib.

Comment (vatsa287) [reply csi/volumeutils.py]: Done

File: csi/volumeutils.py (hunk 1 above)
Comment (vatsa287) [reply csi/volumeutils.py:1030]: Done

**Review by amarts — APPROVED** [review]

**Review by leelavg — APPROVED** [review]
LGTM

File: lib/kadalulib.py [hunk 3]
```diff
@@ -77,6 +77,34 @@ def is_gluster_mount_proc_running(volname, mountpoint):
...
+        except socket.error as msg:
+            logging.error(logf("Failed to open socket connection",
+                               error=msg, host=host))
+            continue
+        finally:
+            sock.close()
+    return False
+
+
+def reachable_hosts(hosts):
```
Comment (aravindavk) [inline lib/kadalulib.py]: nit: Function name is plural, may confuse others who read code. Change it to `reachable_host`

Comment (vatsa287) [reply lib/kadalulib.py]: Done

**Review by aravindavk — APPROVED** [review]

**Pull request merged**